dollarskip.c

```diff
@@ -1,13 +1,25 @@
 #include <stdlib.h>
 #include <string.h>
-char command[8096] = {0};
 int main(int argc, char **argv)
 {
-for (int i=1; i<argc; i++){
-    strcat(strcat(command,argv[i])," ");
-    
-}
-command[strlen(command) - 1] = '\0';
-
-system(command);
+	char command[8096] = {0};
+	if(argc>1) {
+		char shell[10], *env=getenv("SHELL"); //get the env var 'SHELL'
+		if(strcmp(env, "")) { //if env != ""
+			strcpy(shell, env); //copy env to shell
+			strcat(shell, " -c \'"); //add " -c '"
+			strcat(command, shell); //add it to command
+			for (int i=1; i<argc; i++){
```

Comment:
```suggestion
			for (int i=1; i<argc; ++i){
```

Reply:
You have raised a very nice issue.

Here is what i++ looks like in C++

```
(gdb) disassemble foo
Dump of assembler code for function foo:
   0x00000000004004b4 <+0>:     push   %rbp
   0x00000000004004b5 <+1>:     mov    %rsp,%rbp
   0x00000000004004b8 <+4>:     movl   $0x0,-0x8(%rbp)
   0x00000000004004bf <+11>:    mov    -0x8(%rbp),%eax
   0x00000000004004c2 <+14>:    mov    %eax,-0x4(%rbp)
   0x00000000004004c5 <+17>:    addl   $0x1,-0x8(%rbp)
   0x00000000004004c9 <+21>:    pop    %rbp
   0x00000000004004ca <+22>:    retq
End of assembler dump.

```

And heres ++i


```

Reply:
Which is better? I don't know asm well 😄

Reply:
Neither do I really.
However what I do know is the time saved using whatever is better is so redundant that in the time of me pressing the next key ` ` I would have used more time that it would saved.

Reply:
++i will increment the value of i, and then return the incremented value.

     i = 1;
     j = ++i;
     (i is 2, j is 2)

    i++ will increment the value of i, but return the original value that i held before being incremented.

     i = 1;
     j = i++;
     (i is 2, j is 1)




Thats the only information I can find

https://stackoverflow.com/questions/24853/what-is-the-difference-between-i-and-i?answertab=active#tab-top

Reply:
Therefore I can't see any issue with either since we are not adding like in the example.

Reply:
I'm leaving it as `i++`, is that ok @bl-ue, @CleanMachine1?

Reply:
Fine by me